### PR TITLE
Encapsulate change calculations in new ChangeStatistic class.

### PR DESCRIPTION
**Pull request by johnjohndoe**
# Description
- I added the `ChangeStatistic` class to encapsulate computations with regards to lecture change statistics.
- Tests are migrated from `FahrplanMiscTest` to `ChangeStatisticTest` and extended somewhat.

### CODE REVIEW

File: app/src/main/java/nerd/tuxmobil/fahrplan/congress/changes/ChangeStatistic.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    fun getNewLecturesCount() = lectures
+            .count { it.changedIsNew }
+            .also { log("$it new lectures") }
+
+    /**
+     * Returns how many favorites are marked as [canceled][Lecture.changedIsCanceled],
+     * [changed][Lecture.isChanged] or [new][Lecture.changedIsNew].
+     */
+    fun getChangedFavoritesCount() = lectures
+            .count { (it.changedIsCanceled || it.isChanged || it.changedIsNew) && it.highlight }
```
[the remark this answers is not in the document]
Comment (johnjohndoe) [reply app/src/main/java/nerd/tuxmobil/fahrplan/congress/changes/ChangeStatistic.kt]: Fine with me. Changed in 9bc4d97ae1654b127c28eeb00af4fffea6a9cf9c.

File: app/src/test/java/nerd/tuxmobil/fahrplan/congress/changes/ChangeStatisticTest.kt [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+import com.google.common.truth.Truth.assertThat
+import info.metadude.android.eventfahrplan.commons.logging.Logging
+import nerd.tuxmobil.fahrplan.congress.models.Lecture
+import org.junit.Test
+
+class ChangeStatisticTest {
+
+    companion object {
+
+        // region Changed
```
[the remark this answers is not in the document]
Comment (johnjohndoe) [reply app/src/test/java/nerd/tuxmobil/fahrplan/congress/changes/ChangeStatisticTest.kt]: :bird: are still alive but the test is much more readable now. Thank you. See 9bc4d97ae1654b127c28eeb00af4fffea6a9cf9c.

File: app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/MainActivity.java [hunk 3]
```diff
@@ -335,14 +336,13 @@ void showChangesDialog() {
...
+            ChangeStatistic statistic = new ChangeStatistic(changedLectures);
             DialogFragment changesDialog = ChangesDialog.newInstance(
                     scheduleVersion,
-                    FahrplanMisc.getChangedLectureCount(changedLectures, false),
-                    FahrplanMisc.getNewLectureCount(changedLectures, false),
-                    FahrplanMisc.getCancelledLectureCount(changedLectures, false),
-                    FahrplanMisc.getChangedLectureCount(changedLectures, true) +
-                            FahrplanMisc.getNewLectureCount(changedLectures, true) +
-                            FahrplanMisc.getCancelledLectureCount(changedLectures, true),
+                    statistic.getChangedLecturesCount(),
```
[the remark this answers is not in the document]
Comment (johnjohndoe) [reply app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/MainActivity.java]: I agree. Thanks for spotting this. Changed in 634aa9a505e1fcc63c2c585806e4ed6788147dde.

File: app/src/main/java/nerd/tuxmobil/fahrplan/congress/changes/ChangeStatistic.kt [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+    /**
+     * Returns how many lectures are marked as [changed][Lecture.isChanged].
+     */
+    fun getChangedLecturesCount() = lectures
+            .count { it.isChanged }
+            .also { log("$it changed lectures") }
+
+    /**
+     * Returns how many lectures have are marked as [new][Lecture.changedIsNew].
```
Comment (ideadapt) [inline app/src/main/java/nerd/tuxmobil/fahrplan/congress/changes/ChangeStatistic.kt]: typo: -have

File: app/src/test/java/nerd/tuxmobil/fahrplan/congress/changes/ChangeStatisticTest.kt [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    @Test
+    fun `getCanceledLecturesCount returns 0 for an empty list`() {
+        val statistic = ChangeStatistic(emptyList())
+        assertThat(statistic.getCanceledLecturesCount()).isEqualTo(0)
+    }
+
+    @Test
+    fun `getCanceledLecturesCount returns the canceled lectures count`() {
+        val statistic = ChangeStatistic(unchangedLectures + canceledLectures)
+        assertThat(statistic.getCanceledLecturesCount()).isEqualTo(2)
```
Comment (ideadapt) [inline app/src/test/java/nerd/tuxmobil/fahrplan/congress/changes/ChangeStatisticTest.kt]: you could use `.isEqualTo(canceledLectures.size())` as that would match the test description more precisely and is less fragile to changes, but still accurate. (same applies to other tests).

File: app/src/main/java/nerd/tuxmobil/fahrplan/congress/changes/ChangeStatistic.kt (hunk 4 above)
Comment (johnjohndoe) [reply app/src/main/java/nerd/tuxmobil/fahrplan/congress/changes/ChangeStatistic.kt]: Fixed in e5337468b8d9380c77a4931e6dd748e7410eb429.

File: app/src/test/java/nerd/tuxmobil/fahrplan/congress/changes/ChangeStatisticTest.kt (hunk 5 above)
Comment (johnjohndoe) [reply app/src/test/java/nerd/tuxmobil/fahrplan/congress/changes/ChangeStatisticTest.kt]: Improved in e5337468b8d9380c77a4931e6dd748e7410eb429.

**Comment by johnjohndoe** [comment]
@ideadapt I got some feedback from the Travis CI support with regards to the broken GitHub integration. They are currently **hitting GitHub API limits** and they are still trying to resolve this. I expect them to take a while for this. Meanwhile, I now **temporarily** disabled the Travis CI check.

I like to ask you to run `./gradlew clean test` for **this branch** to have a **second machine verifying the complete test suite** for all build flavors. I did that myself:

- `./gradlew clean test` -> :heavy_check_mark: No errors

If it succeeds on your machine, too - please merge.

**Comment by ideadapt** [comment]
All tests green here to.

**Pull request merged**